src/internal/observe/observe.go

```diff
@@ -151,7 +151,17 @@ func Message(ctx context.Context, msgs ...any) {
 	plain := strings.Join(plainSl, " ")
 	loggable := strings.Join(loggableSl, " ")
 
-	logger.Ctx(ctx).Info(loggable)
+	MessageWithStructure(ctx, plain, loggable)
+}
+
+// MessageWithStructure is used to display a progress message as well as a
+// structured log.
+func MessageWithStructure(
+	ctx context.Context,
+	msg, logMsg string,
+	logArgs ...any,
```

Comment:
I don't believe this func is necessary.  `observe.Message` should already be set up to handle hiding arbitrary strings values.  Ie:

```
// instead of this
observe.MessageWithStructure(
  ctx,
  "updating maintenance user/host to "+currentOwner,
  "updating maintenance user/host",
  "new_maintenance_owner",
  clues.Hide(currentOwner))

// do this
observe.MessageWithStructure(
  ctx,
  "updating maintenance user/host to",
  clues.Hide(currentOwner))
  
// or, if you want to add structure explicitly, in addition to the message
observe.MessageWithStructure(
  clues.Add(ctx, "new_maintenance_owner", clues.Hide(currentOwner)),
  "updating maintenance user/host to",
  clues.Hide(currentOwner))

Reply:
mmm part of that was so we could have structured logging of stuff as well. It's kind of sixes though as long as data is hidden